File: src/test/java/org/micoli/php/builders/PostToggleScriptBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.runnableTask.PostToggleScript;

public class PostToggleScriptBuilder {
    private final PostToggleScript script;

    private PostToggleScriptBuilder() {
        this.script = new PostToggleScript();
    }

    public static PostToggleScriptBuilder create() {
        return new PostToggleScriptBuilder();
    }

    public PostToggleScriptBuilder withId(String id) {
        script.id = id;
        return this;
    }

    public PostToggleScriptBuilder withLabel(String label) {
        script.label = label;
        return this;
    }

    public PostToggleScriptBuilder withSource(String source) {
        script.source = source;
        return this;
    }

    public PostToggleScriptBuilder withExtension(String extension) {
        script.extension = extension;
        return this;
    }

    public PostToggleScript build() {
        return script;
    }
}

File: src/test/kotlin/org/micoli/php/builders/PostToggleScriptBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.runnableTask.postToggle.PostToggleScript

class PostToggleScriptBuilder private constructor() {
    private val script: PostToggleScript = PostToggleScript()

    fun withId(id: String): PostToggleScriptBuilder {
        script.id = id
        return this
    }

    fun withLabel(label: String): PostToggleScriptBuilder {
        script.label = label
        return this
    }

    fun withSource(source: String): PostToggleScriptBuilder {
        script.source = source
        return this
    }

    fun withExtension(extension: String): PostToggleScriptBuilder {
        script.extension = extension
        return this
    }

    fun build(): PostToggleScript = script

    companion object {
        fun create(): PostToggleScriptBuilder = PostToggleScriptBuilder()
    }
}
